builtin.c

```diff
@@ -87,15 +87,40 @@ XS(XS_builtin_false)
     XSRETURN_NO;
 }
 
+XS(XS_builtin_inf);
+XS(XS_builtin_inf)
+{
+    dXSARGS;
+    if(items)
+        croak_xs_usage(cv, "");
+    mPUSHn(PL_inf.nv);
```

Comment:
Huh, don't we need EXTEND(SP, 1) before PUSH? mPUSHn says the stack must have room for the item being pushed. Or is there some other magic going on?

Reply:
I guess... the stack always has 1 spot free for simple returns but anything beyond that you have to EXTEND for?

Reply:
Oops, we do need that; but apparently I was just lucky in all the tests here that it didn't matter. But yes, I should add it.

Reply:
Is there any test coverage for this? Here and elsewhere? If it's possible to accidentally break the stack, shouldn't we test that we haven't?

Reply:
I couldn't find a test to trip it up, even compiling with -Accflags=-DSTRESS_REALLOC

Reply:
This kind of thing is quite hard to reliably test for, because you need to ensure that the stack already has exactly zero elements spare when you enter the function. It just grows over time as different things end up using it, so it's quite hard to arrange a size already.

Reply:
Actually, the existing true/false also have a bug. They should be fixed first - see https://github.com/Perl/perl5/pull/21888

Reply:
Is there any existing linting that could be extended to check for this sort of thing?